File: backend/src/InsuraTech.API/Controllers/LifePlansController.cs
using InsuraTech.Application.LifePlans.DTOs;
using InsuraTech.Application.LifePlans.Queries.CalculateLifePlan;
using InsuraTech.Application.LifePlans.Queries.GetLifePlans;
using MediatR;
using Microsoft.AspNetCore.Mvc;

namespace InsuraTech.API.Controllers;

[ApiController]
[Route("api/v1/life-plans")]
[Produces("application/json")]
public sealed class LifePlansController : ControllerBase
{
    private readonly IMediator _mediator;

    public LifePlansController(IMediator mediator) => _mediator = mediator;

    /// <summary>Lista todos los planes de vida disponibles con sus beneficios.</summary>
    [HttpGet]
    [ProducesResponseType(typeof(IReadOnlyList<LifePlanDto>), StatusCodes.Status200OK)]
    public async Task<IActionResult> GetAll(CancellationToken cancellationToken)
    {
        var result = await _mediator.Send(new GetLifePlansQuery(), cancellationToken);
        return Ok(result);
    }

    /// <summary>Valida edad del asegurado y retorna la cotización del plan de vida seleccionado.</summary>
    [HttpGet("calculate")]
    [ProducesResponseType(typeof(LifePlanCalculationDto), StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
    public async Task<IActionResult> Calculate(
        [FromQuery] string planId,
        [FromQuery] DateOnly birthDate,
        CancellationToken cancellationToken)
    {
        var result = await _mediator.Send(
            new CalculateLifePlanQuery { PlanId = planId, BirthDate = birthDate },
            cancellationToken);

        return Ok(result);
    }
}

File: backend_v2/src/main/java/com/insuratech/api/controllers/LifePlansController.java
// Spec: SPEC-012 — planes de vida
package com.insuratech.api.controllers;

import com.insuratech.application.lifeplans.LifePlanService;
import com.insuratech.application.lifeplans.dto.LifePlanCalculationDto;
import com.insuratech.application.lifeplans.dto.LifePlanDto;
import lombok.RequiredArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDate;
import java.util.List;

@RestController
@RequestMapping("/v1/life-plans")
@RequiredArgsConstructor
public class LifePlansController {

    private final LifePlanService lifePlanService;

    @GetMapping
    public ResponseEntity<List<LifePlanDto>> getAll() {
        return ResponseEntity.ok(lifePlanService.getAll());
    }

    @GetMapping("/calculate")
    public ResponseEntity<LifePlanCalculationDto> calculate(
            @RequestParam String planId,
            @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate birthDate) {
        return ResponseEntity.ok(lifePlanService.calculate(planId, birthDate));
    }
}
